core/src/test/java/test/failedreporter/issue2517/FailedDataProviderWithFactoryReporterTest.java

```diff
@@ -0,0 +1,59 @@
+package test.failedreporter.issue2517;
+
+import org.testng.Assert;
+import org.testng.TestNG;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.testng.collections.Lists;
+import test.SimpleBaseTest;
+
+import java.io.File;
+import java.util.List;
+
+public class FailedDataProviderWithFactoryReporterTest extends SimpleBaseTest {
```

Comment:
Please add a reference to this test in `src/test/resources/testng.xml` so that this test gets executed for every build.

Reply:
done